### PR TITLE
Aggregate automm results

### PR DESCRIPTION
**Pull request by suzhoum**
Issue #, if available:

Description of changes:

Supports results aggregation for `multimodal` module.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/autogluon/bench/frameworks/multimodal/exec.py [hunk 1]
```diff
@@ -13,12 +14,24 @@
...
+def _flatten_dict(dd, separator="_", prefix=""):
+    return (
+        {
+            f"{prefix}{separator}{k}" if prefix else k: v
+            for kk, vv in dd.items()
+            for k, v in _flatten_dict(vv, separator, kk).items()
+        }
+        if isinstance(dd, dict)
+        else {prefix: dd}
+    )
```
Comment (gidler) [inline src/autogluon/bench/frameworks/multimodal/exec.py:32]: I had to read this a few times to make sure I understood how it worked. Does this accomplish the same thing, and do you think its more readable?

```python
def _flatten_dict(data, separator='_', prefix=''):
    flattened = {}
    for key, value in data.items():
        if isinstance(value, dict):
            flattened.update(_flatten_dict(value, separator, prefix + key + separator))
        else:
            flattened[prefix + key] = value
    return flattened
```

File: README.md [hunk 2]
```diff
@@ -185,6 +185,8 @@ This will create a new file on S3 with this signature:
 s3://{METRICS_BUCKET}/aggregated/{module}/{benchmark_name}/results_automlbenchmark_{constraint}_{benchmark_name}.csv
 ```
 
+Currently, aggregation is also supported for multimodal benchmark results without the `--constratint` option.
+
```
Comment (gidler) [inline README.md:189]: Do you think we will ever have the idea of constraints for multimodal?

File: src/autogluon/bench/frameworks/multimodal/exec.py (hunk 1 above)
Comment (suzhoum) [reply src/autogluon/bench/frameworks/multimodal/exec.py:32]: I like your refactor and I think they do the same thing. Will update!

File: README.md (hunk 2 above)
Comment (suzhoum) [reply README.md:189]: Yes I'm planning to add constraint to multimodal, but ideally I want to have it look like the same for multimodal and tabular. So I'm still thinking about the design.

**Review by gidler — APPROVED** [review]

**Pull request merged**